compiler/test/dotty/tools/dotc/config/ScalaSettingsTests.scala

```diff
@@ -96,5 +101,100 @@ class ScalaSettingsTests:
+    wconf.map(_.action(warning))
+
+  @Test def `WConf src filter silences warnings from a matching path for virtual file`: Unit =
+    val result = wconfSrcFilterTest(
+      argsStr = "-Wconf:src=path/.*:s",
+      warning = reporting.Diagnostic.Warning(
+        "A warning".toMessage,
+        util.SourcePosition(
+          source = util.SourceFile.virtual(new URI("file:///some/path/file.scala"), ""),
+          span = util.Spans.Span(1L)
+        )
+      )
+    )
+    assertEquals(result, Right(reporting.Action.Silent))
+
+  @Test def `WConf src filter doesn't silence warnings from a non-matching path`: Unit =
+    val result = wconfSrcFilterTest(
+      argsStr = "-Wconf:src=another/.*:s",
+      warning = reporting.Diagnostic.Warning(
+        "A warning".toMessage,
+        util.SourcePosition(
+          source = util.SourceFile.virtual(new URI("file:///some/path/file.scala"), ""),
+          span = util.Spans.Span(1L)
+        )
+      )
+    )
+    assertEquals(result, Right(reporting.Action.Warning))
+
+  @Test def `WConf src filter silences warnings from a matching path for real file`: Unit =
+    val result = Using.resource(Files.createTempFile("myfile", ".scala").nn) { file =>
+      wconfSrcFilterTest(
+        argsStr = "-Wconf:src=myfile.*?\\.scala:s",
+        warning = reporting.Diagnostic.Warning(
+          "A warning".toMessage,
+          util.SourcePosition(
+            source = util.SourceFile(new PlainFile(Path(file)), "UTF-8"),
+            span = util.Spans.Span(1L)
+          )
+        )
+      )
+    }(Files.deleteIfExists(_))
+    assertEquals(result, Right(reporting.Action.Silent))
+
+  @Test def `WConf src filter doesn't silence warnings from a non-matching path for real file`: Unit =
+    val result = Using.resource(Files.createTempFile("myfile", ".scala").nn) { file =>
+      wconfSrcFilterTest(
+        argsStr = "-Wconf:src=another.*?\\.scala:s",
+        warning = reporting.Diagnostic.Warning(
+          "A warning".toMessage,
+          util.SourcePosition(
+            source = util.SourceFile(new PlainFile(Path(file)), "UTF-8"),
+            span = util.Spans.Span(1L)
+          )
+        )
+      )
+    }(Files.deleteIfExists(_))
+    assertEquals(result, Right(reporting.Action.Warning))
+
+  @Test def `WConf src filter reports an error on an invalid regex`: Unit =
+    val result = wconfSrcFilterTest(
+      argsStr = """-Wconf:src=\:s""",
+      warning = reporting.Diagnostic.Warning(
+        "A warning".toMessage,
+        util.SourcePosition(
+          source = util.SourceFile.virtual(new URI("file:///some/path/file.scala"), ""),
+          span = util.Spans.Span(1L)
+        )
+      ),
+    )
+    assertTrue(
+      result.left.exists(errors =>
+        errors.sizeIs == 1 && errors.headOption.exists(_.startsWith("invalid pattern"))
+      )
+    )
+
+  @Test def `WConf src filter can be mixed with other filters with rightmost taking precedence`: Unit =
```

Comment:
Hmm, in Scala 2 the leftmost takes precedence. Also `-Wconf:help` (on Scala 3) says otherwise.

https://github.com/lampepfl/dotty/blob/a095115b43398feeac3ee4329edb87f87da00012/compiler/src/dotty/tools/dotc/config/ScalaSettings.scala#L268-L269

So this is a bug that should be fixed. I think a separate PR is better, so I filed https://github.com/lampepfl/dotty/issues/19885.